### DISCUSSION TITLE
Weighted Nearest Neighbor Analysis for only 2 CITE-seq Abs

### DISCUSSION THREAD

**Discussion by XinLi-0419** [satijalab/seurat#4304] [Q&A] [answered]
Hi, I am trying to learn the Weighted Nearest Neighbor Analysis Vignette and analyze our CITE-seq data. But since we only used two CITE-seq antibodies, I cannot really get another dimensional reduction for the "ADT" in the Vignette. Thus I don't know how to apply the FindMultiModalNeighbors function, for which I think two dimensional reductions are needed. (the runPCA does not work for my two Abs sequencing results). Does anyone know any other way to get around the second dimensional reduction and use the Abs reads directly for wnn? Thanks a lot!

![image](https://user-images.githubusercontent.com/79279783/112913437-95bcc480-90c7-11eb-9b66-253a4880c39a.png)

**Comment by timoast** [comment, marked as answer]
Answered here https://github.com/satijalab/seurat/issues/4301

**Reply by BenSolomon** [reply]
Hi @timoast and @yuhanH - I hope you don't mind if I reopen this question. The the linked answer includes: 

> The WNN method doesn't work for the small number of antibody panel. You may concatenate the two normalized adts to the RNA data.

I'm wondering what the appropriate way would be to accomplish this concatenation. The concatenation itself is relatively simple, but the part that seems tricky is actually getting that concatenation back into a Seurat object, particularly when considering the `scale.data` layer. 

For instance, I can do something like this:

```
combined_count <- rbind(GetAssayData(srt, assay = "SCT", layer = "count"),
                        GetAssayData(srt, assay = "Protein", layer = "count"))
combined_data <- rbind(GetAssayData(srt, assay = "SCT", layer = "data"),
                       GetAssayData(srt, assay = "Protein", layer = "data"))
combined_scale_data <- rbind(GetAssayData(srt, assay = "SCT", layer = "scale.data"),
                             GetAssayData(srt, assay = "Protein", layer = "scale.data"))
new_assay <- CreateAssay5Object(count = combined_count, data = combined_data)
srt[["NEW"]] <- new_assay
```
However, there does not seem to be an option within `CreateAssay5Object` to explicitly specify data for a `scale.data` layer. If I then try to manually assign `combined_scale_data` to the "NEW" assay with 

```
srt_sub[["NEW"]]@scale.data <- combined_scale_data
```

I get the error

```
Error in (function (cl, name, valueClass)  : 
  ‘scale.data’ is not a slot in class “Assay5”
```

And alternatively: 

```
srt_sub[["NEW"]][["scale.data"]] <- combined_scale_data
```

Results in 

```
Error in `[[<-`:
! No feature overlap between new meta data and assay
```

So clearly the SeuratObject class places some restrictions on manually assigning `scale.data` layers. 

Since specifying the `data` layer worked, I could use `ScaleData` to generate a new `scale.data` layer. However:

1. I'm not sure if it is mathematically sound to scale data that contains a mix of normalized RNA counts and normalized ADT counts, as their distributions seem relatively different
2. This prevents me from taking advantage of `SCTransform` for the RNA features. 

### Additional context

My main goal here is to incorporate CD4, CD8, CD45RA, and CD45RO protein expression into the clustering of sorted T-cells into memory populations. As you probably know, these are markers that sperate high level memory T-cell populations quite clearly though CD45RA and CD45RO are splice isoforms of _PTPRC_ that can only be detected by protein expression (though some RNA-based predictive algorithms exist). Gene expression alone does an ok job of segregating these protein markers but they are not super clean and I would actually like the clustering to take the protein markers into account (e.g. give priority to CD4, CD8, CD45RA, and CD45RO protein expression in generating clusters). 

Interesting, I can actually get `FindMultiModalNeighbors` to work with my data, but the results are heavily weighted toward the protein data (e.g. weights > 0.9 for all cells), such that resulting clustering and UMAP reductions effectively look the same as the protein data alone. As a "hack" I can manually update the RNA and Protein weights and this does allow me to start to balance the clustering of RNA and Protein data, but:

1. This hack doesn't seem methodologically sound and
2. As I adjust the balance between RNA and Protein weights, as soon as I start to recovery important gene expression based clusters (e.g. FOXP3+ Tregs), I lose the clear discrimination of CD4 vs CD8 protein or CD45RA vs CD45RO protein. 

**So in summary**, my goal is to generate clusters using GEX and a limited number of ADT features and give a degree of priority to the ADT features. Besides WNN and concatenation, are there other approaches I should consider for this problem?

Thanks!
